// @pulumiverse/doppler/secretssync/awsParameterStore.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AwsParameterStore = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manage an AWS Parameter Store Doppler sync.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 * import * as doppler from "@pulumiverse/doppler";
 *
 * const dopplerParameterStore = new aws.iam.Role("doppler_parameter_store", {
 *     name: "doppler_parameter_store",
 *     assumeRolePolicy: JSON.stringify({
 *         Version: "2012-10-17",
 *         Statement: [{
 *             Effect: "Allow",
 *             Action: "sts:AssumeRole",
 *             Principal: {
 *                 AWS: "arn:aws:iam::299900769157:user/doppler-integration-operator",
 *             },
 *             Condition: {
 *                 StringEquals: {
 *                     "sts:ExternalId": "<YOUR_WORKPLACE_SLUG>",
 *                 },
 *             },
 *         }],
 *     }),
 *     inlinePolicies: [{
 *         name: "doppler_secret_manager",
 *         policy: JSON.stringify({
 *             Version: "2012-10-17",
 *             Statement: [{
 *                 Action: [
 *                     "ssm:PutParameter",
 *                     "ssm:LabelParameterVersion",
 *                     "ssm:DeleteParameter",
 *                     "ssm:RemoveTagsFromResource",
 *                     "ssm:GetParameterHistory",
 *                     "ssm:AddTagsToResource",
 *                     "ssm:GetParametersByPath",
 *                     "ssm:GetParameters",
 *                     "ssm:GetParameter",
 *                     "ssm:DeleteParameters",
 *                 ],
 *                 Effect: "Allow",
 *                 Resource: "*",
 *             }],
 *         }),
 *     }],
 * });
 * const prod = new doppler.integration.AwsParameterStore("prod", {
 *     name: "Production",
 *     assumeRoleArn: dopplerParameterStore.arn,
 * });
 * const backendProd = new doppler.secretssync.AwsParameterStore("backend_prod", {
 *     integration: prod.id,
 *     project: "backend",
 *     config: "prd",
 *     region: "us-east-1",
 *     path: "/backend/",
 *     secureString: true,
 *     tags: {
 *         myTag: "enabled",
 *     },
 *     deleteBehavior: "leave_in_target",
 * });
 * ```
 */
class AwsParameterStore extends pulumi.CustomResource {
    /**
     * Get an existing AwsParameterStore resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AwsParameterStore(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AwsParameterStore.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AwsParameterStore.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["config"] = state ? state.config : undefined;
            resourceInputs["deleteBehavior"] = state ? state.deleteBehavior : undefined;
            resourceInputs["integration"] = state ? state.integration : undefined;
            resourceInputs["kmsKeyId"] = state ? state.kmsKeyId : undefined;
            resourceInputs["path"] = state ? state.path : undefined;
            resourceInputs["project"] = state ? state.project : undefined;
            resourceInputs["region"] = state ? state.region : undefined;
            resourceInputs["secureString"] = state ? state.secureString : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.config === undefined) && !opts.urn) {
                throw new Error("Missing required property 'config'");
            }
            if ((!args || args.integration === undefined) && !opts.urn) {
                throw new Error("Missing required property 'integration'");
            }
            if ((!args || args.path === undefined) && !opts.urn) {
                throw new Error("Missing required property 'path'");
            }
            if ((!args || args.project === undefined) && !opts.urn) {
                throw new Error("Missing required property 'project'");
            }
            if ((!args || args.region === undefined) && !opts.urn) {
                throw new Error("Missing required property 'region'");
            }
            resourceInputs["config"] = args ? args.config : undefined;
            resourceInputs["deleteBehavior"] = args ? args.deleteBehavior : undefined;
            resourceInputs["integration"] = args ? args.integration : undefined;
            resourceInputs["kmsKeyId"] = args ? args.kmsKeyId : undefined;
            resourceInputs["path"] = args ? args.path : undefined;
            resourceInputs["project"] = args ? args.project : undefined;
            resourceInputs["region"] = args ? args.region : undefined;
            resourceInputs["secureString"] = args ? args.secureString : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(AwsParameterStore.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AwsParameterStore = AwsParameterStore;
/** @internal */
AwsParameterStore.__pulumiType = 'doppler:secretsSync/awsParameterStore:AwsParameterStore';
