// @pulumiverse/doppler/secretssync/circleci.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Circleci = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manage a CircleCI Doppler sync.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as doppler from "@pulumiverse/doppler";
 *
 * const prod = new doppler.integration.Circleci("prod", {
 *     name: "Production",
 *     apiToken: "my_api_token",
 * });
 * const backendProd = new doppler.secretssync.Circleci("backend_prod", {
 *     integration: prod.id,
 *     project: "backend",
 *     config: "prd",
 *     resourceType: "project",
 *     resourceId: "github/myorg/myproject",
 *     organizationSlug: "myorg",
 *     deleteBehavior: "leave_in_target",
 * });
 * ```
 */
class Circleci extends pulumi.CustomResource {
    /**
     * Get an existing Circleci resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Circleci(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Circleci.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Circleci.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["config"] = state ? state.config : undefined;
            resourceInputs["deleteBehavior"] = state ? state.deleteBehavior : undefined;
            resourceInputs["integration"] = state ? state.integration : undefined;
            resourceInputs["organizationSlug"] = state ? state.organizationSlug : undefined;
            resourceInputs["project"] = state ? state.project : undefined;
            resourceInputs["resourceId"] = state ? state.resourceId : undefined;
            resourceInputs["resourceType"] = state ? state.resourceType : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.config === undefined) && !opts.urn) {
                throw new Error("Missing required property 'config'");
            }
            if ((!args || args.integration === undefined) && !opts.urn) {
                throw new Error("Missing required property 'integration'");
            }
            if ((!args || args.organizationSlug === undefined) && !opts.urn) {
                throw new Error("Missing required property 'organizationSlug'");
            }
            if ((!args || args.project === undefined) && !opts.urn) {
                throw new Error("Missing required property 'project'");
            }
            if ((!args || args.resourceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceId'");
            }
            if ((!args || args.resourceType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceType'");
            }
            resourceInputs["config"] = args ? args.config : undefined;
            resourceInputs["deleteBehavior"] = args ? args.deleteBehavior : undefined;
            resourceInputs["integration"] = args ? args.integration : undefined;
            resourceInputs["organizationSlug"] = args ? args.organizationSlug : undefined;
            resourceInputs["project"] = args ? args.project : undefined;
            resourceInputs["resourceId"] = args ? args.resourceId : undefined;
            resourceInputs["resourceType"] = args ? args.resourceType : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Circleci.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Circleci = Circleci;
/** @internal */
Circleci.__pulumiType = 'doppler:secretsSync/circleci:Circleci';
