// @pulumiverse/doppler/secretssync/flyio.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Flyio = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manage a Fly.io Doppler sync.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as doppler from "@pulumiverse/doppler";
 *
 * const prod = new doppler.integration.Flyio("prod", {
 *     name: "TF Fly.io",
 *     apiKey: "fo1_xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
 * });
 * const backendProd = new doppler.secretssync.Flyio("backend_prod", {
 *     integration: prod.id,
 *     project: "backend",
 *     config: "prd",
 *     appId: "my-app",
 *     restartMachines: true,
 *     deleteBehavior: "leave_in_target",
 * });
 * ```
 */
class Flyio extends pulumi.CustomResource {
    /**
     * Get an existing Flyio resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Flyio(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Flyio.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Flyio.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["appId"] = state ? state.appId : undefined;
            resourceInputs["config"] = state ? state.config : undefined;
            resourceInputs["deleteBehavior"] = state ? state.deleteBehavior : undefined;
            resourceInputs["integration"] = state ? state.integration : undefined;
            resourceInputs["project"] = state ? state.project : undefined;
            resourceInputs["restartMachines"] = state ? state.restartMachines : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.appId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'appId'");
            }
            if ((!args || args.config === undefined) && !opts.urn) {
                throw new Error("Missing required property 'config'");
            }
            if ((!args || args.integration === undefined) && !opts.urn) {
                throw new Error("Missing required property 'integration'");
            }
            if ((!args || args.project === undefined) && !opts.urn) {
                throw new Error("Missing required property 'project'");
            }
            if ((!args || args.restartMachines === undefined) && !opts.urn) {
                throw new Error("Missing required property 'restartMachines'");
            }
            resourceInputs["appId"] = args ? args.appId : undefined;
            resourceInputs["config"] = args ? args.config : undefined;
            resourceInputs["deleteBehavior"] = args ? args.deleteBehavior : undefined;
            resourceInputs["integration"] = args ? args.integration : undefined;
            resourceInputs["project"] = args ? args.project : undefined;
            resourceInputs["restartMachines"] = args ? args.restartMachines : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Flyio.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Flyio = Flyio;
/** @internal */
Flyio.__pulumiType = 'doppler:secretsSync/flyio:Flyio';
