// @pulumiverse/doppler/secretssync/githubActions.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.GithubActions = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manage a GitHub Actions Doppler sync.
 */
class GithubActions extends pulumi.CustomResource {
    /**
     * Get an existing GithubActions resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new GithubActions(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of GithubActions.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === GithubActions.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["config"] = state ? state.config : undefined;
            resourceInputs["deleteBehavior"] = state ? state.deleteBehavior : undefined;
            resourceInputs["environmentName"] = state ? state.environmentName : undefined;
            resourceInputs["integration"] = state ? state.integration : undefined;
            resourceInputs["orgScope"] = state ? state.orgScope : undefined;
            resourceInputs["project"] = state ? state.project : undefined;
            resourceInputs["repoName"] = state ? state.repoName : undefined;
            resourceInputs["syncTarget"] = state ? state.syncTarget : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.config === undefined) && !opts.urn) {
                throw new Error("Missing required property 'config'");
            }
            if ((!args || args.integration === undefined) && !opts.urn) {
                throw new Error("Missing required property 'integration'");
            }
            if ((!args || args.project === undefined) && !opts.urn) {
                throw new Error("Missing required property 'project'");
            }
            if ((!args || args.syncTarget === undefined) && !opts.urn) {
                throw new Error("Missing required property 'syncTarget'");
            }
            resourceInputs["config"] = args ? args.config : undefined;
            resourceInputs["deleteBehavior"] = args ? args.deleteBehavior : undefined;
            resourceInputs["environmentName"] = args ? args.environmentName : undefined;
            resourceInputs["integration"] = args ? args.integration : undefined;
            resourceInputs["orgScope"] = args ? args.orgScope : undefined;
            resourceInputs["project"] = args ? args.project : undefined;
            resourceInputs["repoName"] = args ? args.repoName : undefined;
            resourceInputs["syncTarget"] = args ? args.syncTarget : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(GithubActions.__pulumiType, name, resourceInputs, opts);
    }
}
exports.GithubActions = GithubActions;
/** @internal */
GithubActions.__pulumiType = 'doppler:secretsSync/githubActions:GithubActions';
