// @pulumiverse/doppler/secretssync/terraformCloud.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TerraformCloud = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * ## Example Usage
 */
class TerraformCloud extends pulumi.CustomResource {
    /**
     * Get an existing TerraformCloud resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new TerraformCloud(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of TerraformCloud.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TerraformCloud.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["config"] = state ? state.config : undefined;
            resourceInputs["deleteBehavior"] = state ? state.deleteBehavior : undefined;
            resourceInputs["integration"] = state ? state.integration : undefined;
            resourceInputs["nameTransform"] = state ? state.nameTransform : undefined;
            resourceInputs["project"] = state ? state.project : undefined;
            resourceInputs["syncTarget"] = state ? state.syncTarget : undefined;
            resourceInputs["variableSetId"] = state ? state.variableSetId : undefined;
            resourceInputs["variableSyncType"] = state ? state.variableSyncType : undefined;
            resourceInputs["workspaceId"] = state ? state.workspaceId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.config === undefined) && !opts.urn) {
                throw new Error("Missing required property 'config'");
            }
            if ((!args || args.integration === undefined) && !opts.urn) {
                throw new Error("Missing required property 'integration'");
            }
            if ((!args || args.nameTransform === undefined) && !opts.urn) {
                throw new Error("Missing required property 'nameTransform'");
            }
            if ((!args || args.project === undefined) && !opts.urn) {
                throw new Error("Missing required property 'project'");
            }
            if ((!args || args.syncTarget === undefined) && !opts.urn) {
                throw new Error("Missing required property 'syncTarget'");
            }
            if ((!args || args.variableSyncType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'variableSyncType'");
            }
            resourceInputs["config"] = args ? args.config : undefined;
            resourceInputs["deleteBehavior"] = args ? args.deleteBehavior : undefined;
            resourceInputs["integration"] = args ? args.integration : undefined;
            resourceInputs["nameTransform"] = args ? args.nameTransform : undefined;
            resourceInputs["project"] = args ? args.project : undefined;
            resourceInputs["syncTarget"] = args ? args.syncTarget : undefined;
            resourceInputs["variableSetId"] = args ? args.variableSetId : undefined;
            resourceInputs["variableSyncType"] = args ? args.variableSyncType : undefined;
            resourceInputs["workspaceId"] = args ? args.workspaceId : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(TerraformCloud.__pulumiType, name, resourceInputs, opts);
    }
}
exports.TerraformCloud = TerraformCloud;
/** @internal */
TerraformCloud.__pulumiType = 'doppler:secretsSync/terraformCloud:TerraformCloud';
