// @pulumiverse/doppler/integration/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TerraformCloud = exports.Flyio = exports.Circleci = exports.AwsSecretsManager = exports.AwsParameterStore = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AwsParameterStore = null;
utilities.lazyLoad(exports, ["AwsParameterStore"], () => require("./awsParameterStore"));
exports.AwsSecretsManager = null;
utilities.lazyLoad(exports, ["AwsSecretsManager"], () => require("./awsSecretsManager"));
exports.Circleci = null;
utilities.lazyLoad(exports, ["Circleci"], () => require("./circleci"));
exports.Flyio = null;
utilities.lazyLoad(exports, ["Flyio"], () => require("./flyio"));
exports.TerraformCloud = null;
utilities.lazyLoad(exports, ["TerraformCloud"], () => require("./terraformCloud"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "doppler:integration/awsParameterStore:AwsParameterStore":
                return new exports.AwsParameterStore(name, undefined, { urn });
            case "doppler:integration/awsSecretsManager:AwsSecretsManager":
                return new exports.AwsSecretsManager(name, undefined, { urn });
            case "doppler:integration/circleci:Circleci":
                return new exports.Circleci(name, undefined, { urn });
            case "doppler:integration/flyio:Flyio":
                return new exports.Flyio(name, undefined, { urn });
            case "doppler:integration/terraformCloud:TerraformCloud":
                return new exports.TerraformCloud(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("doppler", "integration/awsParameterStore", _module);
pulumi.runtime.registerResourceModule("doppler", "integration/awsSecretsManager", _module);
pulumi.runtime.registerResourceModule("doppler", "integration/circleci", _module);
pulumi.runtime.registerResourceModule("doppler", "integration/flyio", _module);
pulumi.runtime.registerResourceModule("doppler", "integration/terraformCloud", _module);
