// @pulumiverse/doppler/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.types = exports.secretssync = exports.projectmember = exports.integration = exports.config = exports.Webhook = exports.ServiceToken = exports.ServiceAccountToken = exports.ServiceAccount = exports.Secret = exports.Provider = exports.ProjectRole = exports.Project = exports.GroupMembers = exports.GroupMember = exports.Group = exports.getUserOutput = exports.getUser = exports.getSecretsOutput = exports.getSecrets = exports.Environment = exports.BranchConfig = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
exports.BranchConfig = null;
utilities.lazyLoad(exports, ["BranchConfig"], () => require("./branchConfig"));
exports.Environment = null;
utilities.lazyLoad(exports, ["Environment"], () => require("./environment"));
exports.getSecrets = null;
exports.getSecretsOutput = null;
utilities.lazyLoad(exports, ["getSecrets", "getSecretsOutput"], () => require("./getSecrets"));
exports.getUser = null;
exports.getUserOutput = null;
utilities.lazyLoad(exports, ["getUser", "getUserOutput"], () => require("./getUser"));
exports.Group = null;
utilities.lazyLoad(exports, ["Group"], () => require("./group"));
exports.GroupMember = null;
utilities.lazyLoad(exports, ["GroupMember"], () => require("./groupMember"));
exports.GroupMembers = null;
utilities.lazyLoad(exports, ["GroupMembers"], () => require("./groupMembers"));
exports.Project = null;
utilities.lazyLoad(exports, ["Project"], () => require("./project"));
exports.ProjectRole = null;
utilities.lazyLoad(exports, ["ProjectRole"], () => require("./projectRole"));
exports.Provider = null;
utilities.lazyLoad(exports, ["Provider"], () => require("./provider"));
exports.Secret = null;
utilities.lazyLoad(exports, ["Secret"], () => require("./secret"));
exports.ServiceAccount = null;
utilities.lazyLoad(exports, ["ServiceAccount"], () => require("./serviceAccount"));
exports.ServiceAccountToken = null;
utilities.lazyLoad(exports, ["ServiceAccountToken"], () => require("./serviceAccountToken"));
exports.ServiceToken = null;
utilities.lazyLoad(exports, ["ServiceToken"], () => require("./serviceToken"));
exports.Webhook = null;
utilities.lazyLoad(exports, ["Webhook"], () => require("./webhook"));
// Export sub-modules:
const config = require("./config");
exports.config = config;
const integration = require("./integration");
exports.integration = integration;
const projectmember = require("./projectmember");
exports.projectmember = projectmember;
const secretssync = require("./secretssync");
exports.secretssync = secretssync;
const types = require("./types");
exports.types = types;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "doppler:index/branchConfig:BranchConfig":
                return new exports.BranchConfig(name, undefined, { urn });
            case "doppler:index/environment:Environment":
                return new exports.Environment(name, undefined, { urn });
            case "doppler:index/group:Group":
                return new exports.Group(name, undefined, { urn });
            case "doppler:index/groupMember:GroupMember":
                return new exports.GroupMember(name, undefined, { urn });
            case "doppler:index/groupMembers:GroupMembers":
                return new exports.GroupMembers(name, undefined, { urn });
            case "doppler:index/project:Project":
                return new exports.Project(name, undefined, { urn });
            case "doppler:index/projectRole:ProjectRole":
                return new exports.ProjectRole(name, undefined, { urn });
            case "doppler:index/secret:Secret":
                return new exports.Secret(name, undefined, { urn });
            case "doppler:index/serviceAccount:ServiceAccount":
                return new exports.ServiceAccount(name, undefined, { urn });
            case "doppler:index/serviceAccountToken:ServiceAccountToken":
                return new exports.ServiceAccountToken(name, undefined, { urn });
            case "doppler:index/serviceToken:ServiceToken":
                return new exports.ServiceToken(name, undefined, { urn });
            case "doppler:index/webhook:Webhook":
                return new exports.Webhook(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("doppler", "index/branchConfig", _module);
pulumi.runtime.registerResourceModule("doppler", "index/environment", _module);
pulumi.runtime.registerResourceModule("doppler", "index/group", _module);
pulumi.runtime.registerResourceModule("doppler", "index/groupMember", _module);
pulumi.runtime.registerResourceModule("doppler", "index/groupMembers", _module);
pulumi.runtime.registerResourceModule("doppler", "index/project", _module);
pulumi.runtime.registerResourceModule("doppler", "index/projectRole", _module);
pulumi.runtime.registerResourceModule("doppler", "index/secret", _module);
pulumi.runtime.registerResourceModule("doppler", "index/serviceAccount", _module);
pulumi.runtime.registerResourceModule("doppler", "index/serviceAccountToken", _module);
pulumi.runtime.registerResourceModule("doppler", "index/serviceToken", _module);
pulumi.runtime.registerResourceModule("doppler", "index/webhook", _module);
pulumi.runtime.registerResourcePackage("doppler", {
    version: utilities.getVersion(),
    constructProvider: (name, type, urn) => {
        if (type !== "pulumi:providers:doppler") {
            throw new Error(`unknown provider type ${type}`);
        }
        return new exports.Provider(name, undefined, { urn });
    },
});
