// @pulumi/digitalocean/genaiAgent.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.GenaiAgent = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
class GenaiAgent extends pulumi.CustomResource {
    /**
     * Get an existing GenaiAgent resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new GenaiAgent(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of GenaiAgent.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === GenaiAgent.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["agentGuardrails"] = state ? state.agentGuardrails : undefined;
            resourceInputs["anthropicApiKeys"] = state ? state.anthropicApiKeys : undefined;
            resourceInputs["anthropicKeyUuid"] = state ? state.anthropicKeyUuid : undefined;
            resourceInputs["apiKeyInfos"] = state ? state.apiKeyInfos : undefined;
            resourceInputs["apiKeys"] = state ? state.apiKeys : undefined;
            resourceInputs["chatbotIdentifiers"] = state ? state.chatbotIdentifiers : undefined;
            resourceInputs["chatbots"] = state ? state.chatbots : undefined;
            resourceInputs["childAgents"] = state ? state.childAgents : undefined;
            resourceInputs["createdAt"] = state ? state.createdAt : undefined;
            resourceInputs["deployments"] = state ? state.deployments : undefined;
            resourceInputs["description"] = state ? state.description : undefined;
            resourceInputs["functions"] = state ? state.functions : undefined;
            resourceInputs["ifCase"] = state ? state.ifCase : undefined;
            resourceInputs["instruction"] = state ? state.instruction : undefined;
            resourceInputs["k"] = state ? state.k : undefined;
            resourceInputs["knowledgeBaseUuids"] = state ? state.knowledgeBaseUuids : undefined;
            resourceInputs["knowledgeBases"] = state ? state.knowledgeBases : undefined;
            resourceInputs["maxTokens"] = state ? state.maxTokens : undefined;
            resourceInputs["modelUuid"] = state ? state.modelUuid : undefined;
            resourceInputs["models"] = state ? state.models : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["openAiApiKeys"] = state ? state.openAiApiKeys : undefined;
            resourceInputs["openAiKeyUuid"] = state ? state.openAiKeyUuid : undefined;
            resourceInputs["parentAgents"] = state ? state.parentAgents : undefined;
            resourceInputs["projectId"] = state ? state.projectId : undefined;
            resourceInputs["provideCitations"] = state ? state.provideCitations : undefined;
            resourceInputs["region"] = state ? state.region : undefined;
            resourceInputs["retrievalMethod"] = state ? state.retrievalMethod : undefined;
            resourceInputs["routeCreatedAt"] = state ? state.routeCreatedAt : undefined;
            resourceInputs["routeCreatedBy"] = state ? state.routeCreatedBy : undefined;
            resourceInputs["routeName"] = state ? state.routeName : undefined;
            resourceInputs["routeUuid"] = state ? state.routeUuid : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["temperature"] = state ? state.temperature : undefined;
            resourceInputs["templates"] = state ? state.templates : undefined;
            resourceInputs["topP"] = state ? state.topP : undefined;
            resourceInputs["updatedAt"] = state ? state.updatedAt : undefined;
            resourceInputs["url"] = state ? state.url : undefined;
            resourceInputs["userId"] = state ? state.userId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.instruction === undefined) && !opts.urn) {
                throw new Error("Missing required property 'instruction'");
            }
            if ((!args || args.modelUuid === undefined) && !opts.urn) {
                throw new Error("Missing required property 'modelUuid'");
            }
            if ((!args || args.projectId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'projectId'");
            }
            if ((!args || args.region === undefined) && !opts.urn) {
                throw new Error("Missing required property 'region'");
            }
            resourceInputs["agentGuardrails"] = args ? args.agentGuardrails : undefined;
            resourceInputs["anthropicApiKeys"] = args ? args.anthropicApiKeys : undefined;
            resourceInputs["anthropicKeyUuid"] = args ? args.anthropicKeyUuid : undefined;
            resourceInputs["apiKeyInfos"] = args ? args.apiKeyInfos : undefined;
            resourceInputs["apiKeys"] = args ? args.apiKeys : undefined;
            resourceInputs["chatbotIdentifiers"] = args ? args.chatbotIdentifiers : undefined;
            resourceInputs["chatbots"] = args ? args.chatbots : undefined;
            resourceInputs["childAgents"] = args ? args.childAgents : undefined;
            resourceInputs["createdAt"] = args ? args.createdAt : undefined;
            resourceInputs["deployments"] = args ? args.deployments : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["functions"] = args ? args.functions : undefined;
            resourceInputs["ifCase"] = args ? args.ifCase : undefined;
            resourceInputs["instruction"] = args ? args.instruction : undefined;
            resourceInputs["k"] = args ? args.k : undefined;
            resourceInputs["knowledgeBaseUuids"] = args ? args.knowledgeBaseUuids : undefined;
            resourceInputs["knowledgeBases"] = args ? args.knowledgeBases : undefined;
            resourceInputs["maxTokens"] = args ? args.maxTokens : undefined;
            resourceInputs["modelUuid"] = args ? args.modelUuid : undefined;
            resourceInputs["models"] = args ? args.models : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["openAiApiKeys"] = args ? args.openAiApiKeys : undefined;
            resourceInputs["openAiKeyUuid"] = args ? args.openAiKeyUuid : undefined;
            resourceInputs["parentAgents"] = args ? args.parentAgents : undefined;
            resourceInputs["projectId"] = args ? args.projectId : undefined;
            resourceInputs["provideCitations"] = args ? args.provideCitations : undefined;
            resourceInputs["region"] = args ? args.region : undefined;
            resourceInputs["retrievalMethod"] = args ? args.retrievalMethod : undefined;
            resourceInputs["routeCreatedBy"] = args ? args.routeCreatedBy : undefined;
            resourceInputs["routeName"] = args ? args.routeName : undefined;
            resourceInputs["routeUuid"] = args ? args.routeUuid : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["temperature"] = args ? args.temperature : undefined;
            resourceInputs["templates"] = args ? args.templates : undefined;
            resourceInputs["topP"] = args ? args.topP : undefined;
            resourceInputs["url"] = args ? args.url : undefined;
            resourceInputs["userId"] = args ? args.userId : undefined;
            resourceInputs["routeCreatedAt"] = undefined /*out*/;
            resourceInputs["updatedAt"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(GenaiAgent.__pulumiType, name, resourceInputs, opts);
    }
}
exports.GenaiAgent = GenaiAgent;
/** @internal */
GenaiAgent.__pulumiType = 'digitalocean:index/genaiAgent:GenaiAgent';
